provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/ProvisioningDeviceClient.java

```diff
@@ -69,13 +68,20 @@ private ProvisioningDeviceClient(String globalEndpoint, String idScope, Provisio
             throw new IllegalArgumentException("Security provider cannot be null");
         }
 
+        if (certificate == null)
+        {
+            //SRS_ProvisioningDeviceClient_25_004: [ The constructor shall throw IllegalArgumentException if certificate is null. ]
```

Comment:
You don't need these kinds of comments in this codebase anymore. Feel free to delete